src/test/fuzz/torcontrol.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <torcontrol.h>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+class DummyTorControlConnection : public TorControlConnection
+{
+public:
+    DummyTorControlConnection() : TorControlConnection{nullptr}
+    {
+    }
+
+    bool Connect(const std::string&, const ConnectionCB&, const ConnectionCB&)
+    {
+        return true;
+    }
+
+    void Disconnect()
+    {
+    }
+
+    bool Command(const std::string&, const ReplyHandlerCB&)
+    {
+        return true;
+    }
+};
+
+void initialize_torcontrol()
+{
+    InitializeFuzzingContext();
+}
+
+FUZZ_TARGET_INIT(torcontrol, initialize_torcontrol)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+
+    TorController tor_controller;
+    while (fuzzed_data_provider.ConsumeBool()) {
+        TorControlReply tor_control_reply;
+        if (fuzzed_data_provider.ConsumeBool()) {
+            tor_control_reply.code = 250;
+        } else if (fuzzed_data_provider.ConsumeBool()) {
+            tor_control_reply.code = 510;
+        } else {
+            tor_control_reply.code = fuzzed_data_provider.ConsumeIntegral<int>();
+        }
```

Comment:
nit: Are the constants from our source code or from somewhere else? Like https://src-ref.docs.torproject.org/tor/control__cmd_8c_source.html ?

Not sure, but if you want, you can also use `CallOneOf` here:

```suggestion
        CallOneOf (fuzzed_data_provider,
        [&] {
            tor_control_reply.code = 250;
        } ,
        [&]{
            tor_control_reply.code = 510;
        } ,
        [&]{
            tor_control_reply.code = fuzzed_data_provider.ConsumeIntegral<int>();
        });
```

Reply:
Good point. Now using `CallOneOf`.

`250` and `510` are from our source:

```sh
$ git grep reply.code
src/test/fuzz/torcontrol.cpp:                tor_control_reply.code = 250;
src/test/fuzz/torcontrol.cpp:                tor_control_reply.code = 510;
src/test/fuzz/torcontrol.cpp:                tor_control_reply.code = fuzzed_data_provider.ConsumeIntegral<int>();
src/torcontrol.cpp:    if (reply.code == 250) {
src/torcontrol.cpp:    } else if (reply.code == 510) { // 510 Unrecognized command
src/torcontrol.cpp:        LogPrintf("tor: Add onion failed; error code %d\n", reply.code);
src/torcontrol.cpp:    if (reply.code == 250) {
src/torcontrol.cpp:    if (reply.code == 250) {
src/torcontrol.cpp:    if (reply.code == 250) {
```